cmd/client/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	bbgrpc "github.com/jmbarzee/bitbox/grpc"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+var defaultPort = optionalEnvString("BIT_BOX_PORT", "8443")
+
+var defaultAddress = optionalEnvString("BIT_BOX_ADDR", func() string {
+	ip, err := getOutboundIP()
+	if err != nil {
+		panic(err)
+	}
+	return ip.String()
+}())
+
+var root = &cobra.Command{
+	Use:   "bitboxc",
+	Short: "A CLI tool for remote BitBox operations",
+	Long:  "Execute remote linux processes on a BitBox server",
+}
+
+func main() {
+	root.AddCommand(cmdStart)
+	root.AddCommand(cmdStop)
+	root.AddCommand(cmdStatus)
+	root.AddCommand(cmdQuery)
+	if err := root.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+func optionalEnvString(key, defaultValue string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	log.Printf("Missing optional environment variable %s, using default %s", key, defaultValue)
+	return defaultValue
+}
+
+func getOutboundIP() (net.IP, error) {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```

Comment:
that's assuming the client is on a public network
if you're sure you want to go this route, it'd be better to loop through the network interfaces and get your ip from there

Reply:
The way this is ultimately used is as a target for the outbound grpc connection on line 66. I'm not sure a public interface makes more sense than loopback if the client is one the same box as the server -- and using loopback would cut all of this code.  If the client isn't on the same box as the server, there is no sane default -- so loopback or making this required seems logical.  IMO loopback is fine since that is probably how most of the development/testing will be done in this project.